### PR TITLE
applied changes in SplashScreen that allows get ScaleType and backgroundColor from preferences

### PR DESCRIPTION
**Pull request by VitorDPires**
### Platforms affected
Android

### Motivation and Context

Need to set ScaleType of splashscreen, and with this, the backgroundColor is also needed.

### Description

With this changes, only thing you need to do is set the preferences on config.xml.
To make things easier, hexadecimal colors can be used in preferences "#FFFFFF".
example:
preference name="SplashScaleType" value="CENTER_INSIDE"
preference name="SplashBackgroundColor" value="#232323"
if no preferences, FIT_XY and #FFFFFF will be the default.

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by NiklasMerz** [comment]
This looks like a breaking change. Please explain more why this is needed. I don't fully understand the changes.

File: src/android/SplashScreen.java [hunk 1]
```diff
@@ -301,16 +305,9 @@ public void run() {
                 splashImageView.setMinimumWidth(display.getWidth());
 
                 // TODO: Use the background color of the webView's parent instead of using the preference.
-                splashImageView.setBackgroundColor(preferences.getInteger("backgroundColor", Color.BLACK));
+                splashImageView.setBackgroundColor(Color.parseColor(preferences.getString("SplashBackgroundColor", "#FFFFFF")));
```
Comment (erisu) [inline src/android/SplashScreen.java:308]: This should be in a separate PR. Preferably keep a single PR focus on a single change.

Also this changes the default behavior. The default was black. Why change to white.

Also the above todo mentioned to not use from preference. Changes seems to go against the previous desired suggestion.

File: src/android/SplashScreen.java [hunk 2]
```diff
@@ -301,16 +305,9 @@ public void run() {
...
 
-                if (isMaintainAspectRatio()) {
-                    // CENTER_CROP scale mode is equivalent to CSS "background-size:cover"
-                    splashImageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
-                }
-                else {
-                    // FIT_XY scales image non-uniformly to fit into image view.
-                    splashImageView.setScaleType(ImageView.ScaleType.FIT_XY);
-                }
+                splashImageView.setScaleType(ImageView.ScaleType.valueOf(getSplashScaleType()));
```
Comment (erisu) [inline src/android/SplashScreen.java:310]: What happens if the user typed in a wrong key? Maybe catch the thrown error, display a warning and fallback to default??

File: src/android/SplashScreen.java [hunk 3]
```diff
@@ -393,6 +390,7 @@ public void onCancel(DialogInterface dialog) {
 
                 spinnerDialog.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
                 spinnerDialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
+                spinnerDialog.getWindow().setGravity(Gravity.CENTER);
```
Comment (erisu) [inline src/android/SplashScreen.java:393]: What does this have to with the image? Again separate PR is recommended for different target changes.

**Review by erisu — CHANGES_REQUESTED** [review]
I see many various changes that should be split into multiple prs.

It's better to split for change tacking and reviewing.

I also take a look at the suggestion to try and avoid breaking changes so old preference could be warned for depreciation.

File: src/android/SplashScreen.java (hunk 3 above)
Comment (NiklasMerz) [reply src/android/SplashScreen.java:393]: This [PR](https://github.com/apache/cordova-plugin-splashscreen/pull/234) is already about this.
